src/Notes/Collection/AbstractNote.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Pinterest for WooCommerce Abstract Note.
+ *
+ * @package Pinterest_For_WooCommerce/Classes/
+ * @version x.x.x
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Pinterest\Notes\Collection;
+
+use Automattic\WooCommerce\Admin\Notes\Note;
+use Automattic\WooCommerce\Admin\Notes\Notes;
+
+use stdClass;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * AbstractNote class. This is a proxy that helps us using the WC Admin Note
+ * class in the context of Pinterest For WooCommerce plugin.
+ *
+ * @since x.x.x
+ */
+abstract class AbstractNote {
+
+	/**
+	 * Check if the note has been previously added.
+	 *
+	 * @throws NotesUnavailableException Throws exception when notes are unavailable.
+	 */
+	public static function note_exists() {
+		$data_store = Notes::load_data_store();
```

Comment:
To help IDEs, we should have the following comment added here:

```suggestion
		/** @var DataStore $data_store */
		$data_store = Notes::load_data_store();
```

And the appropriate `use` statement at the top of the file:

```php
use Automattic\WooCommerce\Admin\Notes\DataStore;
```

Reply:
Cool.

Reply:
👍

Reply:
@JPry It forced me to add a short description to the docblock:
```PHP
	        /**
		 * Notes Data Store.
		 *
		 * @var DataStore $data_store
		 */
```

Is there a less verbose format for type-hinting in the IDE?